src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/Contracts/RestoreRequestParams.cs

```diff
@@ -140,6 +141,26 @@ internal IEnumerable<string> SelectedBackupSets
                 SetOptionValue(RestoreOptionsHelper.SelectedBackupSets, value);
             }
         }
+
+        /// <summary>
+        /// The executation mode for the operation. default is execution
+        /// </summary>
+        public TaskExecutionMode TaskExecutionMode { get; set; }
+
+        /// <summary>
+        /// Target Database name
+        /// </summary>
+        public string DatabaseName
+        {
+            get
+            {
+                return TargetDatabaseName;
```

Comment:
I added the DatabaseName to the request base class assuming all requests will have a database name field. TargetDatabase is the same as DatabaseName if restore so that's why I'm implementing it this way

---
In reply to: [134058351](https://github.com/Microsoft/sqltoolsservice/pull/438#discussion_r134058351) [](ancestors = 134058351)